salt/modules/network.py

```diff
@@ -1887,3 +1895,55 @@ def iphexval(ip):
     a = ip.split('.')
     hexval = ['%02X' % int(x) for x in a]  # pylint: disable=E1321
     return ''.join(hexval)
+
+
+def ip_networks(interface=None, include_loopback=False, verbose=False):
+    '''
+    .. versionaddeed:: Neon
+
+    Returns a list of IPv4 networks to which the minion belongs.
+
+    interface
+        Restrict results to the specified interface(s). This value can be
+        either a single interface name or a list of interfaces. Globbing is
+        also supported.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' network.list_networks
+        salt '*' network.list_networks interface=docker0
+        salt '*' network.list_networks interface=docker0,enp*
+        salt '*' network.list_networks interface=eth*
+    '''
+    return salt.utils.network.ip_networks(
```

Comment:
Maybe use `__utils__['network.ip_networks']` instead, to make it easier to port the utils functions?

Reply:
Side question: I've always found it a bit strange having functions that simply invoke util functions and only forward arguments. For some time I've been thinking about a module, say `utils.py`, e.g.,

```
def fun(name, *args, **kwargs):
    return __utils__[name](*args, **kwargs)
```

Which gives you access to any util functions (and plenty are quite handy) without having to write almost-dummy module functions like that. What would be your thoughts on this? I can see any arguments against, but you have more experience and let me know if there's anything I'm missing.

Reply:
That could be useful for troubleshooting purposes.

I get where you're coming from with regard to having a function that just invokes a utils function, but a lot of this code comes from before `__utils__` was a thing in execution modules. The utils libraries were mainly created so that code exposed to the user via execution modules, as well as code elsewhere in salt, could both import from common code. Simply moving it to salt/modules/network.py would make it unavailable to code that does not have access to the remote execution modules.

Reply:
I agree, but this codes goes into the develop branch, where `__utils__` is a thing - or are you referring to something different? In this state it's hard (read: impossible) to port this to older versions.

Reply:
Yes, I'm referring to something different. I'm saying that putting the code directly into salt/modules/network.py would make unavailable to code that doesn't have access to the execution modules.

Reply:
This is not what I was suggesting: I wasn't objecting about the code being under utils, but the way the util functions are called from the execution module. Or to put this in code:

Instead of

```python
return salt.utils.network.ip_networks(
```

(which cannot be used when porting the module as-is), might be better to use:

```python
return __utils__['network.ip_networks'](
```

Which make the module portable (when porting the utils under `salt://_utils`). Does this make more sense? `salt.utils.network.ip_networks` is really a major PITA when you can't upgrade to the latest Salt version and you need to patch an utils module/function.

Reply:
Ah, OK. Sorry for the misunderstanding. I 100% agree. I just pushed another commit which changes much of the utils module usage to the `__utils__` dunder.